sql: TestShowQueries failed under stress

SHA: https://github.com/cockroachdb/cockroach/commits/60cb230725d183a274a49993819b68e5184976f1

Parameters:
```
TAGS=
GOFLAGS=
```

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stressrace instead of stress and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
make stress TESTS=TestShowQueries PKG=github.com/cockroachdb/cockroach/pkg/sql TESTTIMEOUT=5m STRESSFLAGS='-stderr=false -maxtime 20m -timeout 10m'
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=886977&tab=buildLog

```
=== RUN   TestShowQueries
W180906 05:47:53.426233 144108 server/status/runtime.go:310  [n?] Could not parse build timestamp: parsing time "" as "2006/01/02 15:04:05": cannot parse "" as "2006"
I180906 05:47:53.433953 144108 server/server.go:849  [n?] monitoring forward clock jumps based on server.clock.forward_jump_check_enabled
I180906 05:47:53.434076 144108 base/addr_validation.go:279  [n?] server certificate addresses: IP=127.0.0.1,::1; DNS=localhost,*.local; CN=node
I180906 05:47:53.434095 144108 base/addr_validation.go:319  [n?] web UI certificate addresses: IP=127.0.0.1,::1; DNS=localhost,*.local; CN=node
I180906 05:47:53.436141 144108 server/config.go:493  [n?] 1 storage engine initialized
I180906 05:47:53.436201 144108 server/config.go:496  [n?] RocksDB cache size: 128 MiB
I180906 05:47:53.436220 144108 server/config.go:496  [n?] store 0: in-memory, size 0 B
I180906 05:47:53.439575 144108 server/node.go:373  [n?] **** cluster a38cc37d-bc05-4822-80cc-dd1091a02286 has been created
I180906 05:47:53.439627 144108 server/server.go:1411  [n?] **** add additional nodes by specifying --join=127.0.0.1:39129
I180906 05:47:53.439790 144108 gossip/gossip.go:407  [n1] NodeDescriptor set to node_id:1 address:<network_field:"tcp" address_field:"127.0.0.1:39129" > attrs:<> locality:<> ServerVersion:<major_val:2 minor_val:0 patch:0 unstable:13 > build_tag:"v2.2.0-alpha.00000000-541-g60cb230" started_at:1536212873439739409 
I180906 05:47:53.448111 144108 storage/store.go:1551  [n1,s1] [n1,s1]: failed initial metrics computation: [n1,s1]: system config not yet available
I180906 05:47:53.448185 144108 server/node.go:476  [n1] initialized store [n1,s1]: disk (capacity=512 MiB, available=512 MiB, used=0 B, logicalBytes=6.9 KiB), ranges=1, leases=0, queries=0.00, writes=0.00, bytesPerReplica={p10=7103.00 p25=7103.00 p50=7103.00 p75=7103.00 p90=7103.00 pMax=7103.00}, writesPerReplica={p10=0.00 p25=0.00 p50=0.00 p75=0.00 p90=0.00 pMax=0.00}
I180906 05:47:53.448224 144108 storage/stores.go:242  [n1] read 0 node addresses from persistent storage
I180906 05:47:53.448277 144108 server/node.go:699  [n1] connecting to gossip network to verify cluster ID...
I180906 05:47:53.448351 144153 gossip/gossip.go:941  [n1] gossip connectivity
 
I180906 05:47:53.450088 144108 server/node.go:724  [n1] node connected via gossip and verified as part of cluster "a38cc37d-bc05-4822-80cc-dd1091a02286"
I180906 05:47:53.450113 144108 server/node.go:548  [n1] node=1: started with [<no-attributes>=<in-mem>] engine(s) and attributes []
I180906 05:47:53.450313 144108 server/status/recorder.go:611  [n1] available memory from cgroups (8.0 EiB) exceeds system memory 16 GiB, using system memory
I180906 05:47:53.450339 144108 server/server.go:1802  [n1] Could not start heap profiler worker due to: directory to store profiles could not be determined
I180906 05:47:53.450392 144108 server/server.go:1548  [n1] starting https server at 127.0.0.1:44595 (use: 127.0.0.1:44595)
I180906 05:47:53.450405 144108 server/server.go:1550  [n1] starting grpc/postgres server at 127.0.0.1:39129
I180906 05:47:53.450416 144108 server/server.go:1551  [n1] advertising CockroachDB node at 127.0.0.1:39129
I180906 05:47:53.463424 144084 sql/event_log.go:126  [n1,intExec=optInToDiagnosticsStatReporting] Event: "set_cluster_setting", target: 0, info: {SettingName:diagnostics.reporting.enabled Value:true User:root}
I180906 05:47:53.472793 144371 sql/event_log.go:126  [n1,intExec=set-setting] Event: "set_cluster_setting", target: 0, info: {SettingName:version Value:2.0-13 User:root}
I180906 05:47:53.498365 144376 sql/event_log.go:126  [n1,intExec=disableNetTrace] Event: "set_cluster_setting", target: 0, info: {SettingName:trace.debug.enable Value:false User:root}
I180906 05:47:53.508309 144310 sql/event_log.go:126  [n1,intExec=initializeClusterSecret] Event: "set_cluster_setting", target: 0, info: {SettingName:cluster.secret Value:04f52fee-06c3-44d2-874f-bd531f5730eb User:root}
I180906 05:47:53.513106 144379 sql/event_log.go:126  [n1,intExec=create-default-db] Event: "create_database", target: 50, info: {DatabaseName:defaultdb Statement:CREATE DATABASE IF NOT EXISTS defaultdb User:root}
I180906 05:47:53.515821 144352 sql/event_log.go:126  [n1,intExec=create-default-db] Event: "create_database", target: 51, info: {DatabaseName:postgres Statement:CREATE DATABASE IF NOT EXISTS postgres User:root}
I180906 05:47:53.517280 144108 server/server.go:1604  [n1] done ensuring all necessary migrations have run
I180906 05:47:53.517313 144108 server/server.go:1607  [n1] serving sql connections
I180906 05:47:53.521367 144388 server/server_update.go:67  [n1] no need to upgrade, cluster already at the newest version
I180906 05:47:53.523932 144390 sql/event_log.go:126  [n1] Event: "node_join", target: 1, info: {Descriptor:{NodeID:1 Address:{NetworkField:tcp AddressField:127.0.0.1:39129} Attrs: Locality: ServerVersion:2.0-13 BuildTag:v2.2.0-alpha.00000000-541-g60cb230 StartedAt:1536212873439739409 LocalityAddress:[]} ClusterID:a38cc37d-bc05-4822-80cc-dd1091a02286 StartedAt:1536212873439739409 LastUp:1536212873439739409}
W180906 05:47:53.528521 144108 server/status/runtime.go:310  [n?] Could not parse build timestamp: parsing time "" as "2006/01/02 15:04:05": cannot parse "" as "2006"
I180906 05:47:53.532411 144108 server/server.go:849  [n?] monitoring forward clock jumps based on server.clock.forward_jump_check_enabled
I180906 05:47:53.532524 144108 base/addr_validation.go:279  [n?] server certificate addresses: IP=127.0.0.1,::1; DNS=localhost,*.local; CN=node
I180906 05:47:53.532542 144108 base/addr_validation.go:319  [n?] web UI certificate addresses: IP=127.0.0.1,::1; DNS=localhost,*.local; CN=node
I180906 05:47:53.534204 144108 server/config.go:493  [n?] 1 storage engine initialized
I180906 05:47:53.534226 144108 server/config.go:496  [n?] RocksDB cache size: 128 MiB
I180906 05:47:53.534234 144108 server/config.go:496  [n?] store 0: in-memory, size 0 B
W180906 05:47:53.534277 144108 gossip/gossip.go:1517  [n?] no incoming or outgoing connections
I180906 05:47:53.534317 144108 server/server.go:1413  [n?] no stores bootstrapped and --join flag specified, awaiting init command.
I180906 05:47:53.546914 144384 gossip/client.go:129  [n?] started gossip client to 127.0.0.1:39129
I180906 05:47:53.547290 144469 gossip/server.go:217  [n1] received initial cluster-verification connection from {tcp 127.0.0.1:44307}
I180906 05:47:53.554337 144108 server/node.go:699  [n?] connecting to gossip network to verify cluster ID...
I180906 05:47:53.554390 144108 server/node.go:724  [n?] node connected via gossip and verified as part of cluster "a38cc37d-bc05-4822-80cc-dd1091a02286"
I180906 05:47:53.556946 144432 gossip/gossip.go:941  [n?] gossip connectivity
  n1 [sentinel];
 
I180906 05:47:53.559363 144108 server/node.go:428  [n?] new node allocated ID 2
I180906 05:47:53.559480 144108 gossip/gossip.go:407  [n2] NodeDescriptor set to node_id:2 address:<network_field:"tcp" address_field:"127.0.0.1:44307" > attrs:<> locality:<> ServerVersion:<major_val:2 minor_val:0 patch:0 unstable:13 > build_tag:"v2.2.0-alpha.00000000-541-g60cb230" started_at:1536212873559416212 
I180906 05:47:53.559565 144108 storage/stores.go:242  [n2] read 0 node addresses from persistent storage
I180906 05:47:53.559643 144432 gossip/gossip.go:941  [n2] gossip connectivity
  n1 [sentinel];
  n2 -> n1;
I180906 05:47:53.560339 144153 gossip/gossip.go:941  [n1] gossip connectivity
  n1 [sentinel];
 
I180906 05:47:53.560422 144108 storage/stores.go:261  [n2] wrote 1 node addresses to persistent storage
I180906 05:47:53.560448 144116 storage/stores.go:261  [n1] wrote 1 node addresses to persistent storage
I180906 05:47:53.561706 144108 server/node.go:674  [n2] bootstrapped store [n2,s2]
I180906 05:47:53.561814 144108 server/node.go:548  [n2] node=2: started with [<no-attributes>=<in-mem>] engine(s) and attributes []
I180906 05:47:53.562072 144108 server/status/recorder.go:611  [n2] available memory from cgroups (8.0 EiB) exceeds system memory 16 GiB, using system memory
I180906 05:47:53.562102 144108 server/server.go:1802  [n2] Could not start heap profiler worker due to: directory to store profiles could not be determined
I180906 05:47:53.562171 144108 server/server.go:1548  [n2] starting https server at 127.0.0.1:45419 (use: 127.0.0.1:45419)
I180906 05:47:53.562190 144108 server/server.go:1550  [n2] starting grpc/postgres server at 127.0.0.1:44307
I180906 05:47:53.562206 144108 server/server.go:1551  [n2] advertising CockroachDB node at 127.0.0.1:44307
I180906 05:47:53.563138 144108 server/server.go:1604  [n2] done ensuring all necessary migrations have run
I180906 05:47:53.563167 144108 server/server.go:1607  [n2] serving sql connections
I180906 05:47:53.582021 144090 sql/event_log.go:126  [n2] Event: "node_join", target: 2, info: {Descriptor:{NodeID:2 Address:{NetworkField:tcp AddressField:127.0.0.1:44307} Attrs: Locality: ServerVersion:2.0-13 BuildTag:v2.2.0-alpha.00000000-541-g60cb230 StartedAt:1536212873559416212 LocalityAddress:[]} ClusterID:a38cc37d-bc05-4822-80cc-dd1091a02286 StartedAt:1536212873559416212 LastUp:1536212873559416212}
I180906 05:47:53.590216 144626 sql/event_log.go:126  [n1,client=127.0.0.1:52608,user=root] Event: "create_database", target: 52, info: {DatabaseName:test Statement:CREATE DATABASE IF NOT EXISTS test User:root}
I180906 05:47:53.593736 144626 sql/event_log.go:126  [n1,client=127.0.0.1:52608,user=root] Event: "create_table", target: 53, info: {TableName:test.public.aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa💩aaaaaaaaaaaaaaaaa Statement:CREATE TABLE test.public.aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa💩aaaaaaaaaaaaaaaaa (num INT) User:root}
I180906 05:47:53.616176 144397 rpc/nodedialer/nodedialer.go:92  [intExec=get-table-statistics] connection to n1 established
I180906 05:47:53.628980 144680 rpc/nodedialer/nodedialer.go:92  [n1] connection to n2 established
I180906 05:47:53.653560 144088 server/server_update.go:67  [n2] no need to upgrade, cluster already at the newest version
I180906 05:47:53.654795 144686 util/stop/stopper.go:537  quiescing; tasks left:
1      [async] closedts-subscription
1      [async] closedts-rangefeed-subscriber
I180906 05:47:53.655021 144685 util/stop/stopper.go:537  quiescing; tasks left:
1      [async] closedts-rangefeed-subscriber
I180906 05:47:53.657184 144686 util/stop/stopper.go:537  quiescing; tasks left:
1      [async] closedts-subscription
--- FAIL: TestShowQueries (0.25s)
	show_test.go:570: unexpected query in SHOW QUERIES: "SHOW CLUSTER SETTING version", expected: "SELECT * FROM aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa\u2026"

```

SHA: https://github.com/cockroachdb/cockroach/commits/e17cf16b028f92723fec2ac97bdd816109d056f7

Parameters:
```
TAGS=
GOFLAGS=-race
```

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stressrace instead of stress and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
make stress TESTS=TestShowQueries PKG=github.com/cockroachdb/cockroach/pkg/sql TESTTIMEOUT=5m STRESSFLAGS='-stderr=false -maxtime 20m -timeout 10m'
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=949525&tab=buildLog

```

Goroutine 1162 (running) created at:
  testing.(*T).Run()
      /usr/local/go/src/testing/testing.go:824 +0x564
  testing.runTests.func1()
      /usr/local/go/src/testing/testing.go:1063 +0xa4
  testing.tRunner()
      /usr/local/go/src/testing/testing.go:777 +0x16d
  testing.runTests()
      /usr/local/go/src/testing/testing.go:1061 +0x4e1
  testing.(*M).Run()
      /usr/local/go/src/testing/testing.go:978 +0x2cd
  github.com/cockroachdb/cockroach/pkg/sql_test.TestMain()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/main_test.go:36 +0x15d
  main.main()
      _testmain.go:530 +0x22a

Goroutine 142 (finished) created at:
  github.com/cockroachdb/cockroach/pkg/sql.(*internalExecutorImpl).initConnEx()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/internal.go:201 +0x56d
  github.com/cockroachdb/cockroach/pkg/sql.(*internalExecutorImpl).execInternal()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/internal.go:499 +0x766
  github.com/cockroachdb/cockroach/pkg/sql.(*internalExecutorImpl).queryInternal()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/internal.go:248 +0x141
  github.com/cockroachdb/cockroach/pkg/sql.(*InternalExecutor).Query()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/internal.go:232 +0x10e
  github.com/cockroachdb/cockroach/pkg/server.(*Server).clusterVersion()
      /go/src/github.com/cockroachdb/cockroach/pkg/server/server_update.go:161 +0xcd
  github.com/cockroachdb/cockroach/pkg/server.(*Server).upgradeStatus()
      /go/src/github.com/cockroachdb/cockroach/pkg/server/server_update.go:105 +0x8f
  github.com/cockroachdb/cockroach/pkg/server.(*Server).startAttemptUpgrade.func1()
      /go/src/github.com/cockroachdb/cockroach/pkg/server/server_update.go:63 +0x27b
  github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunAsyncTask.func1()
      /go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:324 +0xf3
==================
I181008 05:53:16.511077 139521 server/server_update.go:67  [n2] no need to upgrade, cluster already at the newest version
I181008 05:53:16.533278 139523 sql/event_log.go:126  [n2] Event: "node_join", target: 2, info: {Descriptor:{NodeID:2 Address:{NetworkField:tcp AddressField:127.0.0.1:41707} Attrs: Locality: ServerVersion:2.0-14 BuildTag:v2.2.0-alpha.00000000-1408-ge17cf16 StartedAt:1538977996160825305 LocalityAddress:[]} ClusterID:6bffd4e4-2943-41ce-9bb0-a40c3e72eb82 StartedAt:1538977996160825305 LastUp:1538977996160825305}
I181008 05:53:16.950474 139026 util/stop/stopper.go:537  quiescing; tasks left:
1      [async] closedts-subscription
1      [async] closedts-rangefeed-subscriber
I181008 05:53:16.956982 139026 util/stop/stopper.go:537  quiescing; tasks left:
1      [async] closedts-subscription
W181008 05:53:16.996015 139260 sql/crdb_internal.go:792  [n1,client=127.0.0.1:55914,user=root] failed to dial into node 2 (LIVE): initial connection heartbeat failed: rpc error: code = Unavailable desc = all SubConns are in TransientFailure, latest connection error: connection error: desc = "transport: Error while dialing dial tcp 127.0.0.1:41707: connect: connection refused"
I181008 05:53:17.000213 139392 util/stop/stopper.go:537  quiescing; tasks left:
1      [async] transport racer
1      [async] closedts-subscription
1      [async] closedts-rangefeed-subscriber
I181008 05:53:17.000698 139162 kv/transport_race.go:113  transport race promotion: ran 26 iterations on up to 372 requests
I181008 05:53:17.001175 139392 util/stop/stopper.go:537  quiescing; tasks left:
1      [async] closedts-subscription
1      [async] closedts-rangefeed-subscriber
I181008 05:53:17.003628 139392 util/stop/stopper.go:537  quiescing; tasks left:
1      [async] closedts-subscription
--- FAIL: TestShowQueries (2.73s)
	show_test.go:596: SELECT node_id, query FROM [SHOW CLUSTER QUERIES]
	show_test.go:596: -- failed to dial into node 2 (LIVE): initial connection heartbeat failed: rpc error: code = Unavailable desc = all SubConns are in TransientFailure, latest connection error: connection error: desc = "transport: Error while dialing dial tcp 127.0.0.1:41707: connect: connection refused"
	testing.go:730: race detected during execution of test

```

--- a/pkg/sql/show_test.go
+++ b/pkg/sql/show_test.go
@@ -485,29 +485,14 @@ func TestShowQueries(t *testing.T) {
 
 	execKnobs := &sql.ExecutorTestingKnobs{}
 
-	tc := serverutils.StartTestCluster(t, 2, /* numNodes */
-		base.TestClusterArgs{
-			ReplicationMode: base.ReplicationManual,
-			ServerArgs: base.TestServerArgs{
-				UseDatabase: "test",
-				Knobs: base.TestingKnobs{
-					SQLExecutor: execKnobs,
-				},
-			},
-		})
-	defer tc.Stopper().Stop(context.TODO())
-
-	conn1 = tc.ServerConn(0)
-	conn2 = tc.ServerConn(1)
-	sqlutils.CreateTable(t, conn1, tableName, "num INT", 0, nil)
-
 	found := false
 	var failure error
+
 	execKnobs.StatementFilter = func(ctx context.Context, stmt string, err error) {
 		if stmt == selectStmt {
 			found = true
 			const showQuery = "SELECT node_id, (now() - start)::FLOAT8, query FROM [SHOW CLUSTER QUERIES]"
-			const showVersionQuery = "SHOW CLUSTER SETTING VERSION"
+			const showVersionQuery = "SHOW CLUSTER SETTING version"
 
 			rows, err := conn1.Query(showQuery)
 			if err != nil {
@@ -561,6 +546,22 @@ func TestShowQueries(t *testing.T) {
 			}
 		}
 	}
+
+	tc := serverutils.StartTestCluster(t, 2, /* numNodes */
+		base.TestClusterArgs{
+			ReplicationMode: base.ReplicationManual,
+			ServerArgs: base.TestServerArgs{
+				UseDatabase: "test",
+				Knobs: base.TestingKnobs{
+					SQLExecutor: execKnobs,
+				},
+			},
+		})
+	defer tc.Stopper().Stop(context.TODO())
+
+	conn1 = tc.ServerConn(0)
+	conn2 = tc.ServerConn(1)
+	sqlutils.CreateTable(t, conn1, tableName, "num INT", 0, nil)
 
 	if _, err := conn2.Exec(selectStmt); err != nil {
 		t.Fatal(err)

